app/src/components/MinMaxSlider.tsx

```diff
@@ -0,0 +1,84 @@
+import { ReactElement,useCallback, useEffect, useRef } from "react";
+import { useAtom, PrimitiveAtom } from "jotai";
+import "./MinMaxSlider.css";
+
+type MinMaxSliderProps = {
+  sliderMin: number;
+  sliderMax: number;
+  minAtom: PrimitiveAtom<number>;
+  maxAtom: PrimitiveAtom<number>;
+};
+export default function MinMaxSlider({sliderMin,sliderMax,minAtom,maxAtom}:MinMaxSliderProps):ReactElement {
+    const [minVal, setMinVal] = useAtom(minAtom);
+    const [maxVal, setMaxVal] = useAtom(maxAtom);
+    const minValRef = useRef(sliderMin);
+    const maxValRef = useRef(sliderMax);
+    const range = useRef<HTMLDivElement>(null);
+    
+    // Convert to percentage
+    const getPercent = useCallback(
+      (value:number) => Math.round(((value - sliderMin) / (sliderMax - sliderMin)) * 100),
+      [ sliderMin,  sliderMax]
+    );
+  
+    // Set width of the range to decrease from the left side **
+    useEffect(() => {
+      const minPercent = getPercent(minVal);
+      const maxPercent = getPercent(maxValRef.current);
+
+  
+      if (range.current) {
+        range.current.style.left = `${minPercent}%`;
+        range.current.style.width = `${maxPercent - minPercent}%`;
+      }
+    }, [minVal, getPercent]);
+  
+    // Set width of the range to decrease from the right side 
+    useEffect(() => {
+      const minPercent = getPercent(minValRef.current);
+      const maxPercent = getPercent(maxVal);
+  
+      if (range.current) {
+        range.current.style.width = `${maxPercent - minPercent}%`;
+      }
+    }, [maxVal, getPercent]);
+  
+    return (
+      <div className="container">
+        <input
+          type="range"
+          min={sliderMin}
+          max={sliderMax}
+          value={minVal}
+          step={sliderMax-sliderMin == 1 ? 0.1 : 1}
+          onChange={(event) => {
+            const value = Math.min(Number(event.target.value), maxVal);
+            setMinVal(value);
+            minValRef.current = value;
+          }}
+          className="thumb thumb--left"
+        />
+        <input
+          type="range"
+          min={sliderMin}
+          max={sliderMax}
+          value={maxVal}
+          step={sliderMax-sliderMin == 1 ? 0.1 : 1}
+          onChange={(event) => {
+            const value = Math.max(Number(event.target.value), minVal);
+            setMaxVal(value);
+            maxValRef.current = value;
+          }}
+          className="thumb thumb--right"
+
+        />
+  
+        <div className="slider">
+          <div className="slider__track" />
+          <div ref={range} className="slider__range" />
```

Comment:
#### Should Fix
The use of `ref` and `useEffect` to sync slider width is smart, but I think it would be simpler to use CSS variables here.
```suggestion
          <div className="slider__range" style={{ "--min-percent": getPercent(minVal), "--max-percent": getPercent(maxVal) }} />
```

The variables can be used in CSS like this:
```css
  .slider__range {
    ...
    left: calc(1% * var(--min-percent));
    width: calc(1% * (var(--max-percent) - var(--min-percent)));
  }
```

Reply:
It is not always minVal and maxVal that get passed into getPercent - please see line 26 and 27. It depends on if the change was made on minVal or maxVal, and to detect that there are the useEffects. Lastly, to style the div in useEffect we need the range ref. 

I am not sure how we can run that logic and assign the result (a number) to a variable and then use it in inline styling. If we could it will look much cleaner for sure. Let me know.

Reply:
@pea-nut-z  I see. So `useEffect` is to detect the changes in `minVal` and `maxVal`.
But wouldn't the div re-render when those values change?
For example, if `minVal` is 0 and `maxVal` is 1, the div will be initially rendered as:
```
<div className="slider__range" style={{ "--min-percent": 0, "--max-percent": 100 }}>
```
and when `minVal` changes to 0.3, I think the div will re-render as:
```
<div className="slider__range" style={{ "--min-percent": 30, "--max-percent": 100 }}>
```
That's why I thought it was possible to make this without `useEffect`.

Reply:
You're correct. I was able to remove a lot of code in this approach. Please take a look.